web/src/main/java/org/mskcc/smile/web/SampleController.java

```diff
@@ -102,6 +103,29 @@ public ResponseEntity<List<SmileSampleIdMapping>> fetchSampleIdMappingsByInputDa
                 .body(sampleIdsList);
     }
 
+    /**
+     * Given a valid inputId, returns smileSample
+     * @param inputId
+     * @return ResponseEntity
+     * @throws Exception
+     */
+    @ApiOperation(value = "Fetch SmileSample by inputId",
+            nickname = "fetchSmileSampleByInputIdGET")
+    @RequestMapping(value = "/sampleById/{inputId}",
+            method = RequestMethod.GET,
+            produces = "application/json")
+    public ResponseEntity<SmileSample> fetchSmileSampleByInputIdGET(
+            @ApiParam(value = "input id to search with", required = true)
+            @PathVariable String inputId) throws Exception {
+        SmileSample smileSample = sampleService.getSampleByInputId(inputId);
```

Comment:
this will return only the information that the `SmileSample` node contains which isn't very informative for Voyager/PMs.

If we return a published smile sample instead that would address the specific needs for both Voyager and PMs.

Reply:
You can call the [sampleService](https://github.com/mskcc/smile-server/blob/master/service/src/main/java/org/mskcc/smile/service/impl/RequestServiceImpl.java#L164) to get the published `PublishedSmileSample` with the `smileSampleId` returned from the query above